lib/liquid/tags/include.rb

```diff
@@ -70,9 +70,15 @@ def render_to_output_buffer(context, output)
 
       old_template_name = context.template_name
       old_partial       = context.partial
+
       begin
-        context.template_name = template_name
-        context.partial       = true
+        context.template_name = if Template.file_system.respond_to?(:actual_template_name)
+          Template.file_system.actual_template_name(template_name).delete_suffix('.liquid')
```

Comment:
The other reason having a Template#name property would be useful is that it could be used as part of given a name to the initial template.  I think we could provide that as a parse option, then Template#render can set `context.template_name = template.name if template.name`.  I think that would make it easy to set the top-level template name so that it can be included in error messages.